ui/packages/catalog-ui-search/src/main/webapp/component/oauth-item/oauth-item.tsx

```diff
@@ -0,0 +1,179 @@
+import styled from 'styled-components'
+const Common = require('../../js/Common.js')
+const user = require('../singletons/user-instance.js')
+
+const Root = styled.div`
+  display: block;
+  position: relative;
+  height: 3 * ${props => props.theme.minimumButtonSize};
+  cursor: pointer;
+`
+
+const Div = styled.div`
+  width: calc(100% - 2.75rem);
+`
+
+const DetailDate = styled.div`
+  padding: 0rem ${props => props.theme.minimumSpacing};
+  height: ${props => props.theme.minimumButtonSize};
+  line-height: ${props => props.theme.minimumButtonSize};
+  display: flex;
+  flex-flow: row wrap;
+  justify-content: flex-end;
+`
+
+const Date = styled.span`
+  left: ${props => props.theme.minimumSpacing};
+  height: inherit;
+  line-height: inherit;
+  text-align: right;
+`
+const Content = styled.div`
+  background: rgba(255, 255, 255, 0.11);
+  cursor: pointer;
+  z-index: 1;
+  position: relative;
+  left: 0px;
+  top: 0px;
+  text-align: center;
+  padding: 0px 0.625rem;
+  display: flex;
+  flex-direction: column;
+`
+
+const TextDiv = styled.div`
+  padding: ${props => props.theme.minimumSpacing} 0rem;
+  flex: 2;
+`
+
+const Text = styled.div`
+  vertical-align: top;
+  overflow: hidden;
+  text-overflow: ellipsis;
+`
+
+const ButtonDiv = styled.div`
+  padding: ${props => props.theme.minimumSpacing} 0rem;
+  flex: 1;
+`
+
+const ButtonText = styled.button`
+  padding: 0rem ${props => props.theme.minimumSpacing};
+`
+
+const Remove = styled.div`
+  width: 2.75rem;
+  height: 100%;
+  position: absolute;
+  right: 0px;
+  top: 0px;
+  overflow: hidden;
+`
+
+const RemoveButton = styled.button`
+  width: ${props => props.theme.minimumButtonSize};
+  height: 100%;
+  transform: translateX(0%);
+  transition: transform ${props => props.theme.coreTransitionTime} linear;
+`
+
+type Props = {
+  model: Backbone.Model
+}
+
+const OauthItem = ({ model }: Props) => {
+  const modelJSON = model.toJSON()
+  const when = Common.getMomentDate(modelJSON.when)
+
+  let buttonText: String
+  if (modelJSON.type === 'login') {
+    buttonText = 'Sign in'
+  } else {
+    buttonText = 'Authorize'
+  }
+
+  let detail: String
+  if (modelJSON.type === 'login') {
+    detail =
+      'Unable to query the ' +
+      modelJSON.sourceId +
+      ' source. Please login or remove it from the query.'
+  } else {
+    detail =
+      'Attempting to query the ' +
+      modelJSON.sourceId +
+      ' source. Do you authorize to query this source? If not please remove the source from the query.'
+  }
+
+  return (
+    <Root>
+      <Div>
+        <DetailDate>
+          <Date>{when.toString()}</Date>
+        </DetailDate>
+        <Content>
+          <TextDiv>
+            <Text>{detail}</Text>
+          </TextDiv>
+          <ButtonDiv>
+            <button
+              className="is-primary"
+              onClick={() => {
+                setTimeout(() => {
+                  model.collection.remove(model)
```

Comment:
❓ What does this line do?

Reply:
It's to remove the notification when the `-` button is clicked